File: backend/metier_generator/templates/threed/middlewares/security/auth/threed_auth.js
// Middleware d’authentification avancée pour le module threed (Node.js)
function threedAuth(req, res, next) {
  if (!req.user || !req.user.isAuthenticated) {
    return res.status(401).json({ error: 'Authentification requise' });
  }
  next();
}

module.exports = threedAuth;

File: backend/metier_generator/templates/threed/middlewares/security/auth/threed_auth.py
"""
Middleware d’authentification avancée pour le module threed.
"""


def authenticate_request(request):
    # Logique d’authentification métier
    user = getattr(request, "user", None)
    if not user or not user.is_authenticated:
        raise PermissionError("Authentification requise")
    return True
